Makefile-az.mk

```diff
@@ -18,7 +18,12 @@ KARPENTER_FEDERATED_IDENTITY_CREDENTIAL_NAME ?= KARPENTER_FID
 CUSTOM_VNET_NAME ?= $(AZURE_CLUSTER_NAME)-vnet
 CUSTOM_SUBNET_NAME ?= nodesubnet
 
-az-all:             az-login az-create-workload-msi az-mkaks-cilium      az-create-federated-cred az-perm               az-perm-acr az-configure-values             az-build az-run          az-run-sample ## Provision the infra (ACR,AKS); build and deploy Karpenter; deploy sample Provisioner and workload
+az-all:             az-login az-create-workload-msi az-mkaks-cilium      az-create-federated-cred az-perm               az-perm-acr az-configure-values             az-build az-run          az-run-sample ## Provision the infra (ACR,AKS); build and deploy Karpenter; deploy sample Provisioner and workload 
+
+az-all-cniv1:             az-login az-create-workload-msi az-mkaks-cniv1      az-create-federated-cred az-perm               az-perm-acr az-configure-values             az-build az-run          az-run-sample ## Provision the infra (ACR,AKS); build and deploy Karpenter; deploy sample Provisioner and workload
+
+az-all-cni-overlay:  az-login az-create-workload-msi az-mkaks-overlay     az-create-federated-cred az-perm               az-perm-acr az-configure-values             az-build az-run          az-run-sample ## Provision the infra (ACR,AKS); build and deploy Karpenter; deploy sample Provisioner and workload
+
```

Comment:
seems we could improve the composability of these. 

Having two shared make steps 
make az-auth: az-login az-create-workload-msi 
make az-configure-and-build: az-perm az-perm-acr az-configure-values  az-build az-run


then we can refactor all of these to 
```yaml 
az-all:  az-auth az-mkaks-cilium   az-configure-and-build ## Provision the infra (ACR,AKS); build and deploy Karpenter; deploy sample Provisioner and workload 

az-all-cniv1:   az-auth az-mkaks-cniv1 az-configure-and-build 

az-all-cni-overlay:  az-auth az-mkaks-overlay  az-configure-and-build
```